slim.Dockerfile

```
@@ -0,0 +1,23 @@
+FROM python:3.7-slim as runtime
+EXPOSE 8080
+
+WORKDIR /girder
+
+RUN apt-get update \
+  && apt-get install -qy --no-install-recommends \
+    git \
+    libldap2-dev \
+    libsasl2-dev \
+  && rm -rf /var/lib/apt/lists/*
+COPY . /girder/
+
+# avoid psutil GCC dependency by using unofficial large_image wheel
+RUN pip install \
+  --no-cache-dir \
+  --upgrade \
+  --find-links https://girder.github.io/large_image_wheels \
+  --upgrade-strategy eager .
+
+COPY --from=girder/girder:latest /usr/share/girder/static/ /usr/local/share/girder/static/
+
+ENTRYPOINT ["girder", "serve"]
```

It will be [easier to override this](https://docs.docker.com/engine/reference/builder/#understand-how-cmd-and-entrypoint-interact) by invoking users if this is`:
```suggestion
CMD ["girder", "serve"]
```

If we foresee wanting to invoke a variety of different `girder ...` commands in typical usage (which I do not), we could use:
```
ENTRYPOINT ["girder"]
CMD ["serve"]
```

---

Docker users currently have to do this at a bare minimum:

``` bash
docker run girder/girder --host 0.0.0.0
```

If we changed the entrypoint to a CMD, every docker deployment of girder would break.

---

One idea, get rid of `--host ...` requirement with: https://github.com/girder/girder/pull/3167/files#diff-3254677a7917c6c01f55212f86c57fbfR30

---

After discussion, I realize it's better to use `ENTRYPOINT`, even if we eliminate the need to pass `--host` by adding a config file. Users are going to typically need to pass `--database`. Using an `ENTRYPOINT` will allow:
```bash
docker run girder/girder --database mongodb://foo/girder
```
instead of
```bash
docker run girder/girder girder serve --database mongodb://foo/girder
```

Realistically, users are not going to need to run `bash` or other commands in this image under normal usage. If they really do need to run something other than `girder serve`, they have the escape hatch of specifying `--entrypoint`.